copier/jinja_ext.py

```diff
@@ -0,0 +1,109 @@
+"""Jinja2 extensions built for Copier."""
+
+from __future__ import annotations
+
+from typing import Any, Callable, Sequence
+
+from jinja2 import nodes
+from jinja2.exceptions import UndefinedError
+from jinja2.ext import Extension
+from jinja2.parser import Parser
+from jinja2.sandbox import SandboxedEnvironment
+
+
+class YieldEnvironment(SandboxedEnvironment):
+    """Jinja2 environment with a `yield_context` attribute.
+
+    This is simple environment class that extends the SandboxedEnvironment
+    for use with the YieldExtension, mainly for avoiding type errors.
+
+    We use the SandboxedEnvironment because we want to minimize the risk of hidden malware
+    in the templates so we use the SandboxedEnvironment instead of the regular one.
+    Of course we still have the post-copy tasks to worry about, but at least
+    they are more visible to the final user.
+    """
+
+    yield_context: dict[str, Any]
```

Comment:
I think that dicts are better suited for variable-length collections, but we know this one will always have two items, so:

```suggestion
    yield_name: str
    yield_iterable: Iterable[Any]
```

This change, of course, implies more changes across the rest of the file, including the docstrings.